### PR TITLE
New sharding rule for batchnorm inference op

### PR DESCRIPTION
**Pull request by sshonTT**
stablehlo.batch_norm_inference had no sharding rule in shardy dialect.

### Ticket
N/A

### Problem description
It fails compiling hardnet tensor parallel model when lowering stablehlo graph, on the op `stablehlo.batch_norm_inference`

### What's changed
Adding sharding rule for this op.

### Checklist
- [v] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by sshonTT** [comment]
> Can you also add a link to the issue in shardy(or pr) if you have it

@tapspatel , Apparently, shardy blocks direct contributions so I posted an issue [here](https://github.com/openxla/shardy/issues/844)

**Comment by sshonTT** [comment]
@nsmithtt, @vwellsTT Hi, can I get your reviews on here for merge it?
